### ISSUE TITLE
Entering a drug that is not in the pre-populated list throws error. 

### ISSUE THREAD

**Issue by nspies13** [genome/civic-server#101]
Attempted to add "Trastuzumab" and "Herceptin" (same drug, different names). Neither were in the pre-populated list, so it would not allow me to submit the evidence item. I think drug should accept free-text in the event a drug does not fit. Did we decide against this? 

Trastuzumab =  https://pubchem.ncbi.nlm.nih.gov/substance/50067185

@jmcmichael - This may be a client-side issue, I'm not sure.

**Comment by malachig** [comment]
I just encountered another case of this. 
https://civic.genome.wustl.edu/#/events/genes/52/summary/variants/270/summary/evidence/689/summary#evidence

Would like to be able to add onartuzumab to that entry but it silently fails when I attempt to do so.  Presumably because this drug is not in PubChem.  Since this is an antibody, I'm not sure if it ever will be added to PubChem?  Maybe.   But in the meantime we are unable to capture predictions for some of the latest (and most exciting) drugs...  We should probably relax this requirement in the submission form.  

@jmcmichael and @acoffman I'm not sure if this actually a requirement of the DB schema or just the curation form?  I don't see a way to add them in the Admin interface either.  If we want to be strict for most users, could we at least open a path to adding new drugs in the admin interface that will then be available in the pop down menu of the evidence item form?

**Comment by acoffman** [comment]
I will double check with @jmcmichael, but I think this is a limitation of the form. There's nothing on the backend that requires a pubchem id.

In the meantime, I went ahead and added "onartuzumab" manually as a new drug and associated it with this evidence item.

**Comment by malachig** [comment]
Thank Adam.  Here are some additional ones where a specific drug 'Cetuximab' is missing and the user entered a less desirable option instead:
https://civic.genome.wustl.edu/#/events/genes/14/summary/variants/272/summary#variant

Can we add 'Cetuximab' as well?

This is another example of an antibody that will never be added to PubChem.  I guess they just don't cover these drugs since they are not 'chemicals' exactly.  Since there are so many 'biologic' drugs this will probably come up some what frequently.  Would it be hard to expose a drug view in the Admin interface so that we could add these manually when needed?  That way it is not totally freeform but allows us to patch these specific issues.

**Comment by acoffman** [comment]
'Cetuximab' is added as a new Drug into the database.

Yeah, I can definitely give you guys a 'drugs' tab in the admin interface where you can add new ones yourselves!

**Comment by malachig** [comment]
Hmmm, still did not let me add 'Cetuximab' one of these evidence items.  It did not appear in the drop down either.

**Comment by malachig** [comment]
Another two that are missing: 'Nivolumab' and 'Pembrolizumab'.  Note to self, these are needed for this evidence item:
https://civic.genome.wustl.edu/#/events/genes/11335/summary/variants/276/summary/evidence/704/summary#evidence

**Comment by malachig** [comment]
Here is another case where we will want to add 'trastuzumab' when we are able to.

https://civic.genome.wustl.edu/#/events/genes/254/summary/variants/278/summary/evidence/706/summary#evidence

**Comment by malachig** [comment]
Need 'CBL0137' to complete this entry:
https://civic.genome.wustl.edu/#/events/genes/3741/summary/variants/298/summary/evidence/744/summary

**Comment by malachig** [comment]
Need 'pazopanib' to complete this entry:
https://civic.genome.wustl.edu/#/events/genes/2073/summary/variants/277/summary/evidence/705/summary#evidence

**Comment by malachig** [comment]
Need 'GSK321' to complete this entry:
https://civic.genome.wustl.edu/#/events/genes/26/summary/variants/59/summary/evidence/732/summary#evidence

**Comment by kkrysiak** [comment]
Need 'cetuximab' to complete this entry:
https://civic.genome.wustl.edu/#/events/genes/3976/summary/variants/290/summary/evidence/730/summary#evidence

**Comment by kkrysiak** [comment]
Nivolumab and Pembrolizumab for this one:
https://civic.genome.wustl.edu/#/events/genes/11335/summary/variants/276/summary/evidence/743/summary#evidence

**Comment by acoffman** [comment]
This is now fixed on the live site! You can add now enter drugs on the 'Add Evidence' form that are not in the autocomplete suggestions. Additionally, there is an "Add Drug" form field on the Utilities tab, and you can edit which drugs are associated with a given evidence item from the "Edit Evidence Item" form.

I believe I have added all the drugs needed to complete these entries as well. I can go in and complete them myself if you would like, but there are a couple I need a little clarification on.

Meanwhile, I've opened #130 to address updating the drug name autocomplete/suggestion so that things that are not in PubChem, but that we do have locally, will appear.

**Comment by acoffman** [comment]
This is out in production.

**Comment by malachig** [comment]
I am still not able to get this to work.  Here are some notes:
- If I try to enter a drug that does not come up in the drop down when I am editing an evidence statement and hit `Submit Revision for Review` nothing happens
- If I do the same as above, but I change something else about the record, I get a successful submission but the drug addition is not part of the revision.

- I was able to use the admin function to add a new drug name, but I can't really tell if that worked.  Can we get a drug tab in the admin interface so that we can see the current state of things?  This will also help us tell which drugs have been entered manually and which came from PubChem and also help us to prevent duplicates that are just spelled slightly differently.

**Comment by malachig** [comment]
Note to self, EGFR OVEREXPRESSION has a lot of examples of drug names that are missing etc.  

Here is a specific example:
https://civic.genome.wustl.edu/#/events/genes/19/summary/variants/193/summary/evidence/783/summary#evidence

Would like to add Nimotuzumab to that one.

**Comment by acoffman** [comment]
I can definitely add an admin tab for drugs, that's easy. I'm not sure what's happening with the drug names - it seems to work in all cases for me, I'm unable to replicate the issue on my machine. I'll see what I can figure out.

**Comment by malachig** [comment]
Maybe we can look at this together in our meeting this afternoon.

**Comment by acoffman** [comment]
I think I've figured it out! Even though the forms look the same, the frontend is actually using a different form for submitting a new evidence item than it is for editing an existing one. The changes that got made to the form to support it on one form were never made for the other one. I was trying to submit a new evidence item (mistakenly assuming they were the same form). 

Going to dig into the frontend and see if I can figure out the fix.

**Comment by acoffman** [comment]
Turned out to be an easy fix. Pushed to master in the client repo.

**Comment by kkrysiak** [comment]
I've submitted edits to address the drugs associated with the evidence items labeled above. We still need to tackle EGFR OVEREXPRESSION and obviously search for other instances we didn't note here.

**Comment by acoffman** [comment]
We are seeing duplicate drug entries appear with identical names when using the edit form (and potentially the add form too)

**Comment by acoffman** [comment]
Source of the duplicate drug names has been identified and fixed - and a merge tool now exists to fix the existing duplicates. Just waiting on puppet to deploy to prod!
